http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/ResourcesServiceInterp.scala

```diff
@@ -0,0 +1,358 @@
+package org.broadinstitute.dsde.workbench.leonardo.http.service
+
+import cats.Parallel
+import cats.effect.Async
+import cats.implicits.toFunctorOps
+import cats.mtl.Ask
+import cats.syntax.all._
+import org.broadinstitute.dsde.workbench.google2.DiskName
+import org.broadinstitute.dsde.workbench.leonardo.{
+  AppContext,
+  AppName,
+  AppStatus,
+  CloudContext,
+  DiskStatus,
+  RuntimeName,
+  WorkspaceId
+}
+import org.broadinstitute.dsde.workbench.leonardo.db.{
+  appQuery,
+  clusterQuery,
+  kubernetesClusterQuery,
+  nodepoolQuery,
+  persistentDiskQuery,
+  DbReference,
+  DiskServiceDbQueries,
+  KubernetesServiceDbQueries
+}
+import org.broadinstitute.dsde.workbench.leonardo.http.{
+  ctxConversion,
+  CreateAppRequest,
+  CreateDiskRequest,
+  CreateRuntimeRequest,
+  CreateRuntimeResponse,
+  GetAppResponse,
+  GetPersistentDiskResponse,
+  GetRuntimeResponse,
+  ListAppResponse,
+  ListPersistentDiskResponse,
+  ListRuntimeResponse2,
+  UpdateDiskRequest,
+  UpdateRuntimeRequest
+}
+import org.broadinstitute.dsde.workbench.leonardo.model.{
+  LeoAuthProvider,
+  NonDeletableDisksInProjectFoundException,
+  NonDeletableRuntimesInProjectFoundException
+}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.UserInfo
+
+import scala.concurrent.ExecutionContext
+
+final class ResourcesServiceInterp[F[_]: Parallel](authProvider: LeoAuthProvider[F])(implicit
+  F: Async[F],
+  dbRef: DbReference[F],
+  ec: ExecutionContext
+) extends ResourcesService[F]
+    with RuntimeService[F]
+    with AppService[F]
+    with DiskService[F] {
+  override def deleteAllResources(userInfo: UserInfo,
+                                  googleProject: GoogleProject,
+                                  deleteInCloud: Boolean,
+                                  deleteDisk: Boolean
+  )(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      _ <-
+        if (deleteInCloud) deleteAllResourcesInCloud(userInfo, googleProject, deleteDisk)
+        else deleteAllResourcesRecords(userInfo, googleProject)
+    } yield ()
+
+  private def deleteAllResourcesInCloud(userInfo: UserInfo, googleProject: GoogleProject, deleteDisk: Boolean)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      _ <- deleteAllRuntimesInCloud(userInfo, googleProject, deleteDisk)
+      _ <- deleteAllAppsInCloud(userInfo, googleProject, deleteDisk)
+      // Delete any potential left over orphaned disk in the project - This may be a complete overkill as the deletion
+      // of most disks done as part of delete runtimes and delete apps will still be in progress at that point
+      _ <- if (deleteDisk) deleteAllDisksInCloud(userInfo, googleProject) else F.unit
```

Comment:
Not sure if we need to do this, it is more of a last minute effort to clean up potentially orphaned disks, but doing this async is weird

Reply:
I think we need to do this because there may be disks in the project that are not attached to a runtime/app. I'm curious if this causes any errors due to double-deletion of disks though (since attached disk deletion will still be in progress)..

Reply:
Or maybe we pass `deleteDisk=false` to `deleteAllRuntimesInCloud / deleteAllAppsInCloud` and just delete all the disks in one shot?

Reply:
💡 That might be a lot cleaner, yes!

Reply:
Thinking about it, the issue with these async calls is that many of the delete disks calls will fail because they might still be attached to the VMs or Apps 🤔 
So we need some sort of polling for deletion before moving forward with the disks

Reply:
Oh yeah, I see. Or maybe:
1. Delete all the apps/runtimes with deleteDisk=true
2. Find all the _unattached_ disks via DB query, and just delete those

Reply:
Yep, sounds like a plan! I bet this won't be the most elegant query ever written but as long as it does the job we are good :)